src/lib/etl.py

```diff
@@ -0,0 +1,110 @@
+"""ETL for SPOC Papers"""
+import spacy  # type: ignore
+import pandas as pd  # type: ignore
+import lxml.etree as etree  # type: ignore
+
+from typing import Any, Dict, List
+
+TEI = {"tei": "http://www.tei-c.org/ns/1.0"}
+
+
+def get_date(xml_doc: etree.Element) -> str:
+    """Extracts Publication Date if it exists"""
+    pub_date_element = xml_doc.find(
+        "tei:teiHeader/tei:fileDesc/tei:publicationStmt/tei:date",
+        namespaces=TEI,  # noqa: E501
+    )
+    if pub_date_element is not None:
+        pub_date = pub_date_element.attrib.get("when", "")
+    else:
+        pub_date = ""
+    return pub_date
+
+
+def get_entities(doc: spacy.tokens.Doc) -> tuple:
+    """Iterate through document's entities and add to lists"""
+    species, places, habitats = [], [], []
+
+    def __add_ent__(entity):
+        record_info = [
+            (token._.canonical, token._.entity_desc) for token in entity
+        ]  # noqa: E501
+        if entity.label_.startswith("SPECIES"):
+            species.extend(record_info)
+        elif entity.label_.startswith("LOCATION"):
+            places.extend(record_info)
+        elif entity.label_.startswith("HABITAT"):
+            habitats.extend([rec[0] for rec in record_info])
+
+    for ent in doc.ents:
+        __add_ent__(ent)
+
+    # TODO: Check doc._.overlap and add to lists
+
+    # Removes duplicates from species, locations, and habitats
+    species = list(set(species))
+    locations = list(set(places))
+    habitats = list(set(habitats))
+    return species, locations, habitats
+
+
+def get_all_text(xml_doc: etree.Element) -> str:
```

Comment:
Does anything call this method (besides the test)?

Reply:
No, removed from latest commit.